### ISSUE TITLE
High cardinality metric are kept forever or until restart of Web

### ISSUE THREAD

**Issue by Typositoire** [concourse/concourse#8571]
## Summary

We have our modified version of oxygen-mask and this create a new pipeline every minute. One of the metric generated by concourse is `concourse_jobs_schedulingDuration_bucket` which is declared as a HistogramVector https://github.com/concourse/concourse/blob/master/atc/metric/emitter/prometheus.go#L216-L225 

Normally it shouldn't be an issue except for the fact that one of the label is `job_id`, this pipeline basically create 10 bucket every minute. The temporary solution was to restart web to start over but here is the details after a 14 days run.

Number of lines per metrics exposed by metric endpoint
```
18461 concourse_jobs_schedulingDuration_count
18461 concourse_jobs_schedulingDuration_sum
258454 concourse_jobs_schedulingDuration_bucket
```

When scraping our agent fails because the payload is 32mb (we use VMAgent, VictoriaMetric drop in replacement)

After a restart, the payload drops to 45kb. 

## Steps to reproduce

- Create 1 pipeline every minute
- Wait for ~2 weeks
- Cry

## Expected results

Metrics endpoint should stop exposing metrics about old build after a few days.

## Actual results

Keeps ALL the metrics until a restart happens

## Additional context

We use VictoriaMetrics stack but it works exactly the same way Prometheus does. It does not impact anything at cause here and the issue should be reproducible using Vanilla Prometheus Stack.

## Triaging info

* Concourse version: 7.8.2
* Browser (if applicable): N/A
* Did this used to work? We just upgrade to >7 and this metric probably didn't exist in v6.

**Comment by Typositoire** [comment]
I can provide payload (metric endpoint) from before and after a restart if necessary.

**Comment by Typositoire** [comment]
As a reference, we have just hit the same issue with Kyverno, but they have a setting called `metricsRefreshInterval`. You can probably get inspired there ;-)

Ref: https://kyverno.io/docs/monitoring/#configuring-the-metrics

**Comment by zaafar** [comment]
> Metrics endpoint should stop exposing metrics about old build after a few days.

Agree, this would be a nice solution for this problem. Any update from concourse on this issue?

**Comment by marco-m** [comment]
The approach taken by Fabio might be an inspiration: https://github.com/fabiolb/fabio/pull/1015
